pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go

```diff
@@ -228,6 +230,31 @@ func getProfileVersion(config *checkconfig.CheckConfig) uint64 {
 	return profileVersion
 }
 
+func getDeviceType(store *metadata.Store) string {
+	deviceType := strings.ToLower(store.GetScalarAsString("device.type"))
+	switch deviceType {
+	case
+		"access_point",
+		"firewall",
+		"load_balancer",
+		"pdu",
+		"printer",
+		"router",
+		"sd-wan",
+		"sensor",
+		"server",
+		"storage",
+		"switch",
+		"ups",
+		"wlc":
+		return deviceType
+	case "":
+		return "other"
+	}
+	log.Debugf("Unsupported device type: %s", deviceType)
+	return "other"
```

Comment:
This looks like a kind of validation, shall we handle that in https://github.com/DataDog/datadog-agent/blob/7b793fe7ecef6c1e47043d39573bfd439946540c/pkg/collector/corechecks/snmp/internal/configvalidation/config_validate_enrich.go ?

Reply:
Maybe we can return an error if the type is unsupported?

Reply:
addressed in [18d6c0b](https://github.com/DataDog/datadog-agent/pull/21791/commits/18d6c0b82c6c8246e6f2001417ee7bff38088740)